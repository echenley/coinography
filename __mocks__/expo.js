const Expo = require.requireActual('expo')

Expo.Font.loadAsync = jest.fn()

module.exports = Expo
