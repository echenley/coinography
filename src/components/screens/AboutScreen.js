// @flow

import React from 'react'
import { StyleSheet } from 'react-native'
import {
  Body,
  Button,
  Card,
  CardItem,
  Container,
  Content,
  Icon,
  Left,
  Text,
  Thumbnail,
} from 'native-base'
import { WebBrowser } from 'expo'
import Me from '../../images/Me.jpeg'

class AboutScreen extends React.Component<{}> {
  static navigationOptions = {
    title: 'About',
  }

  openBlog = () => {
    WebBrowser.openBrowserAsync('http://henleyedition.com')
  }

  openGitHub = () => {
    WebBrowser.openBrowserAsync('https://github.com/echenley')
  }

  openTwitter = () => {
    WebBrowser.openBrowserAsync('https://twitter.com/echenley')
  }

  render() {
    return (
      <Container>
        <Content padder>
          <Card>
            <CardItem bordered>
              <Left>
                <Thumbnail small source={Me} />
                <Body>
                  <Text style={styles.heading}>Evan Henley</Text>
                </Body>
              </Left>
            </CardItem>
            <CardItem bordered>
              <Body>
                <Text>
                  Classical musician turned software engineer. Particularly
                  interested in ambitious React projects, forward-thinking
                  companies building state of the art web apps, and
                  participation in the open source JavaScript community. When
                  I’m not impetuously coding, I’m reading nonfiction, playing
                  convoluted board games, or building glorious fires in the
                  backyard.
                </Text>
              </Body>
            </CardItem>
            <CardItem footer>
              <Button iconLeft onPress={this.openGitHub} small transparent>
                <Icon name="logo-github" style={styles.icon} />
                <Text>GitHub</Text>
              </Button>
              <Button iconLeft onPress={this.openTwitter} small transparent>
                <Icon name="logo-twitter" style={styles.icon} />
                <Text>Twtter</Text>
              </Button>
              <Button iconLeft onPress={this.openBlog} small transparent>
                <Icon name="link" style={styles.icon} />
                <Text>Blog</Text>
              </Button>
            </CardItem>
          </Card>
        </Content>
      </Container>
    )
  }
}

const styles = StyleSheet.create({
  heading: {
    fontWeight: 'bold',
  },
  icon: {
    fontSize: 18,
  },
})

export default AboutScreen
