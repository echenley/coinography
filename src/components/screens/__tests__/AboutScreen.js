jest.mock('expo')

import * as React from 'react'
import { shallow } from 'enzyme'
import { WebBrowser } from 'expo'
import { Button } from 'native-base'
import AboutScreen from '../AboutScreen'

WebBrowser.openBrowserAsync = jest.fn()

const renderComponent = () => shallow(<AboutScreen />)

describe('<AboutScreen />', () => {
  it('renders correctly', () => {
    expect(renderComponent().getElement()).toMatchSnapshot()
  })

  it('opens web browser for links', () => {
    const component = renderComponent()

    expect(WebBrowser.openBrowserAsync).not.toHaveBeenCalled()

    component
      .find(Button)
      .at(0)
      .simulate('press')
    expect(WebBrowser.openBrowserAsync).toHaveBeenCalledTimes(1)
    expect(WebBrowser.openBrowserAsync).toHaveBeenCalledWith(
      'https://github.com/echenley'
    )

    component
      .find(Button)
      .at(1)
      .simulate('press')
    expect(WebBrowser.openBrowserAsync).toHaveBeenCalledTimes(2)
    expect(WebBrowser.openBrowserAsync).toHaveBeenCalledWith(
      'https://twitter.com/echenley'
    )

    component
      .find(Button)
      .at(2)
      .simulate('press')
    expect(WebBrowser.openBrowserAsync).toHaveBeenCalledTimes(3)
    expect(WebBrowser.openBrowserAsync).toHaveBeenCalledWith(
      'http://henleyedition.com'
    )
  })
})
