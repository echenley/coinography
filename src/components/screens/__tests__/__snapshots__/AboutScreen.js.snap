// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutScreen /> renders correctly 1`] = `
<Styled(Container)
  virtual={undefined}
>
  <Styled(Content)
    padder={true}
    virtual={undefined}
  >
    <Styled(Card)
      virtual={undefined}
    >
      <Styled(CardItem)
        bordered={true}
        virtual={undefined}
      >
        <Styled(Left)
          virtual={undefined}
        >
          <Styled(Thumbnail)
            small={true}
            source={
              Object {
                "testUri": "../../../src/images/Me.jpeg",
              }
            }
            virtual={undefined}
          />
          <Styled(Body)
            virtual={undefined}
          >
            <Styled(Text)
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
              virtual={undefined}
            >
              Evan Henley
            </Styled(Text)>
          </Styled(Body)>
        </Styled(Left)>
      </Styled(CardItem)>
      <Styled(CardItem)
        bordered={true}
        virtual={undefined}
      >
        <Styled(Body)
          virtual={undefined}
        >
          <Styled(Text)
            virtual={undefined}
          >
            Classical musician turned software engineer. Particularly interested in ambitious React projects, forward-thinking companies building state of the art web apps, and participation in the open source JavaScript community. When I’m not impetuously coding, I’m reading nonfiction, playing convoluted board games, or building glorious fires in the backyard.
          </Styled(Text)>
        </Styled(Body)>
      </Styled(CardItem)>
      <Styled(CardItem)
        footer={true}
        virtual={undefined}
      >
        <Styled(Button)
          iconLeft={true}
          onPress={[Function]}
          small={true}
          transparent={true}
          virtual={undefined}
        >
          <Styled(Icon)
            name="logo-github"
            style={
              Object {
                "fontSize": 18,
              }
            }
            virtual={undefined}
          />
          <Styled(Text)
            virtual={undefined}
          >
            GitHub
          </Styled(Text)>
        </Styled(Button)>
        <Styled(Button)
          iconLeft={true}
          onPress={[Function]}
          small={true}
          transparent={true}
          virtual={undefined}
        >
          <Styled(Icon)
            name="logo-twitter"
            style={
              Object {
                "fontSize": 18,
              }
            }
            virtual={undefined}
          />
          <Styled(Text)
            virtual={undefined}
          >
            Twtter
          </Styled(Text)>
        </Styled(Button)>
        <Styled(Button)
          iconLeft={true}
          onPress={[Function]}
          small={true}
          transparent={true}
          virtual={undefined}
        >
          <Styled(Icon)
            name="link"
            style={
              Object {
                "fontSize": 18,
              }
            }
            virtual={undefined}
          />
          <Styled(Text)
            virtual={undefined}
          >
            Blog
          </Styled(Text)>
        </Styled(Button)>
      </Styled(CardItem)>
    </Styled(Card)>
  </Styled(Content)>
</Styled(Container)>
`;
